Content.Shared/Atmos/Components/GasMinerComponent.cs

```diff
@@ -0,0 +1,47 @@
+using Content.Shared.Atmos;
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.Atmos.Components;
+
+[NetworkedComponent]
+[AutoGenerateComponentState]
+[RegisterComponent]
+public sealed partial class GasMinerComponent : Component
+{
+    [AutoNetworkedField]
+    [ViewVariables(VVAccess.ReadOnly)]
+    public bool Enabled { get; set; } = true;
+
+    [AutoNetworkedField]
+    [ViewVariables(VVAccess.ReadOnly)]
+    public bool Idle { get; set; } = false;
+
+    /// <summary>
+    ///      If the number of moles in the external environment exceeds this number, no gas will be mined.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField("maxExternalAmount")]
+    public float MaxExternalAmount { get; set; } = float.PositiveInfinity;
+
+    /// <summary>
+    ///      If the pressure (in kPA) of the external environment exceeds this number, no gas will be mined.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField("maxExternalPressure")]
+    public float MaxExternalPressure { get; set; } = Atmospherics.GasMinerDefaultMaxExternalPressure;
+
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField("spawnGas")]
+    public Gas? SpawnGas { get; set; } = null;
```

Comment:
Is there a reason why this is nullable?

Reply:
My first answer is [that's how it's been for three years](https://github.com/space-wizards/space-station-14/blame/master/Content.Server/Atmos/Piping/Other/Components/GasMinerComponent.cs#L30C51-L30C51). But I'm investigating further as well.

Reply:
The full answer is that it makes the object behave slightly better when it's in an invalid state. Because `Gas` is an enum, if you didn't set this field in the yml then it would default to `Oxygen` since that's value zero in the Gas enum. With the nullable the miner will show the disabled state if you forgot to actually provide a gas type in the data files.

Reply:
Probably better to make the datafield `required: true` then.